### ISSUE TITLE
The `@stylistic/less/unit-case` rule passes over the unit Less reads in front of an exponent

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#646]
### What minimal example or steps are needed to reproduce the bug?

A dimension whose number Less does not read as an exponent:

```less
@v: 1E5PX;
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "overrides": [
    {
      "files": ["**/*.less"],
      "customSyntax": "postcss-less",
      "rules": {
        "@stylistic/less/unit-case": "lower"
      }
    }
  ]
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json
```

### What did you expect to happen?

Two warnings, one about `E` and one about `PX`. Less reads no exponent: its reader of a number takes digits and one period, and a letter behind them opens the unit, so `less@4.9.1` compiles that line to `a { width: 1E 5PX; }` — the dimension `1E`, whose unit is `E`, beside the dimension `5PX`. That `E` is a unit and not a keyword is what the arithmetic says: `1E+5PX` compiles to `6E` and `1E-5PX` to `-4E`, the left operand's unit carried through, and a guard `when (2E > 1)` matches.

### What actually happened?

One warning, about `PX` alone:

```
1:8-10  Expected "PX" to be "px"
```

and `--fix` leaves `@v: 1E5px;`, which Less compiles to `a { width: 1E 5px; }` — the `E` still in upper case under `lower`. Under `upper` the same the other way: `@v: 1e5px;` is reported at `1:8-10` and comes out `@v: 1e5PX;`, compiled to `1e 5PX`. `@v: 1E5;`, which Less compiles to `1E 5`, draws nothing at all.

Measured in the six places Less reads a value: a variable's value, a set of media parameters, a `calc()`, a call of another name, a custom property, and the last declaration of a block written without a semicolon. A declaration closing on a semicolon is the one place where nothing is wrong: `1E5PX` spells none of the characters `. # @ $ + / ' " * ` ( { } -`, so Less stores that value unread and prints it whole, and what reaches the browser is the CSS dimension `1E5PX` whose unit really is `PX`. `1E-5PX` spells a hyphen, so it is read as an expression even there, and the rule names `PX` alone in it too.

### What is the cause?

The reading is plain CSS's. `@csstools/css-tokenizer` reads `1E5PX` as one dimension whose number is `1E5` and whose unit is `PX`, and `lightningcss` compiles it to `100000px`; `postcss-value-parser`, which `getDimension` reads the number and the unit with, parts it the same way. Less does not: `entities.dimension` in `parser.js` matches `^([+-]?\d*\.?\d+)(%|[a-z_]+)?`, which has no exponent in it, so the number ends at `1` and the unit is `E`.

This is the other end of the same word as [#527](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/527) and [#633](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/633), which widened where a unit **ends** under this namespace. Here it is where the number ends and the unit **begins**, which `readsUnitAsIdentifier` does not reach: `getDimension` takes the split from `valueParser.unit` and only ever shortens the unit behind it.

### Do you have a proposal to fix the bug?

Give the syntax a say in where the number ends, the way it now has a say in where the unit ends, and under Less end the number at the first character that is no digit and no period. `1E5PX` is then the two dimensions `1E` and `5PX` to name, the way `10PX-2REM` became two under #633, and the remainder is already read again by the walk that #633 put in.

Note that the same reader ends a unit at a digit — `10PX9` compiles to `10PX 9` — where nothing is at stake: the message names `PX9` for a unit Less calls `PX`, and the fix writes the same bytes either way, since a digit has no case. Whether that message is worth narrowing belongs with this decision.
